framework/common/src/main/java/edu/kit/kastel/mcse/ardoco/core/api/Disambiguation.java

```diff
@@ -0,0 +1,160 @@
+/* Licensed under MIT 2023. */
+package edu.kit.kastel.mcse.ardoco.core.api;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.collections.api.map.ImmutableMap;
+import org.eclipse.collections.api.map.MutableMap;
+import org.eclipse.collections.impl.factory.Maps;
+import org.jetbrains.annotations.NotNull;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+
+/**
+ * This class represents an abbreviation with a known set of meanings. An abbreviation is a string such as "ArDoCo" and has the meaning "Architecture
+ * Documentation Consistency". The abbreviation that is disambiguated by the meanings of this class is final, but the meanings can be changed. An instance of
+ * this class can be serialized and deserialized into JSON using Jackson.
+ */
+@JsonSerialize(using = Disambiguation.DisambiguationSerializer.class)
+public class Disambiguation implements Comparable<Disambiguation>, Serializable {
+    private final String abbreviation;
+    private final LinkedHashSet<String> meanings;
```

Comment:
As in `java private final Set<String> meanings`? I think it might be better to be specific. Set makes no guarantee about iteration order by itself and doesn't necessarily give information about mutability either (most people and ides just assume immutability). Thus, I think the interfaces Set and Map shouldn't be used, if we want to comply with the determinism architecture test. If I were to use Set here, it would additionally require annotating the class `@Deterministic` (at least that's my understanding of the annotation). This seems like a long-term maintenance headache to me, since it relies on manual review and technically needs to be checked manually on every change.

I do however recognize the need for immutable/unmodifiable (deterministic) collections, which the interfaces are often used to represent implicitly (which is poor design by Java itself imo, sadly the explicitly unmodifiable collections are package-private in java.util). In such cases, I propose using the classes [UnmodifiableLinkedHashSet](https://github.com/ArDoCo/Core/blob/bd8235ac3c5f9e07531773197cf4b60785e28d3f/framework/common/src/main/java/edu/kit/kastel/mcse/ardoco/core/common/collection/UnmodifiableLinkedHashSet.java) and [UnmodifiableLinkedHashMap](https://github.com/ArDoCo/Core/blob/bd8235ac3c5f9e07531773197cf4b60785e28d3f/framework/common/src/main/java/edu/kit/kastel/mcse/ardoco/core/common/collection/UnmodifiableLinkedHashMap.java) instead. These classes offer deterministic iteration order and explicitly specify mutability. They also implement Set and Map and thus do not break any existing code where Set or Map is used. The alternative solution is using sorted eclipse collections everywhere, which would require implementing comparables and inducing performance penalties.

| Type/Feature  | Mutability | Determinism | Compatible with built-in types |
| ------------- | ------------- | ------------- | ------------- |
| Set/Map | No | No | Yes |
| Eclipse Collections | Yes | No | Only mutable |
| Eclipse Collections Sorted | Yes | Yes | Only mutable |
| (Unmodifiable-)LinkedHashSet/Map | Yes  | Yes | Yes |